### PR TITLE
[CALCITE-3300] Add a star identifier as operand to "SqlCountAggFunction" if no operands given while in the method "createCall"

### PR DESCRIPTION
**Pull request by weidong3630**
Resolve the bug described in issue https://issues.apache.org/jira/browse/CALCITE-3300

### CODE REVIEW

**Comment by weidong3630** [comment]
> exec0 is not necessary, converting sqlNode back and forth is not necessary.

Thanks, I will fix later.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -4038,17 +4058,25 @@ Sql throws_(String errorMessage) {
       }
     }
 
-    String exec() {
+    SqlNode exec0() {
       final Planner planner =
-          getPlanner(null, SqlParser.Config.DEFAULT, schema, config);
+              getPlanner(null, SqlParser.Config.DEFAULT, schema, config);
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: No need for this change

**Comment by jinxing64** [comment]
In current code https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L5132
`*` is explicitly checked and removed for `count(*)` when converting sql to rel. Shall we just replace `*` with the first column of the child ?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (weidong3630) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Fixed.

**Comment by weidong3630** [comment]
@jinxing64 Like @julianhyde said,
Calcite allows multiple arguments to COUNT. It counts a row if none of the arguments are NULL:
> select count(x, y), count(x), count(y), count(*)
> from (values (1, 2),
>    (3, CAST(NULL AS INTEGER))) as t (x, y);
+-----------------+-----------------+-----------------+-----------------+
|     EXPR$0      |     EXPR$1      |     EXPR$2      |     EXPR$3      |
+-----------------+-----------------+-----------------+-----------------+
| 1               | 2               | 1               | 2               |
+-----------------+-----------------+-----------------+-----------------+
It is a natural generalization to make COUNT() identical to COUNT(*). So that's how Calcite represents COUNT(*) internally. 

So the meaning of `count(*)` and counting first column is not always the same because that the element of the first column may be null which will effect the result.

**Pull request closed without merging**